### PR TITLE
Milestone 2: Implemented the CRUD methods

### PR DESCRIPTION
**Pull request by prg-04**
### I have done the following to reach this milestone:

- [ ] Removed hardcoded items in the tasks array
- [ ] Implemented the functions for adding and removing an element from the list
- [ ] Added an Editing function
- [ ] Set the completed property of the List to be toggled to true or false based on whether the taks is completed
- [ ] Dynamically update the index when a task is added or removed

### CODE REVIEW

File: src/store.js [hunk 1]
```diff
@@ -9,19 +9,69 @@ class Store {
...
+
   static removeList(id) {
     const lists = Store.getList();
     const index = lists.findIndex((list) => list.id === id);
-
     if (index !== -1) {
       lists.splice(index, 1);
       localStorage.setItem('list', JSON.stringify(lists));
     }
   }
```
Comment (youmari) [inline src/store.js]: - [ ] When I tried to remove a task the whole application freezes and doesn't respond anymore so please make sure to fix this issue to complete the requirements 
Screenshot 2023-05-12 at 18 48 35

**Review by youmari — CHANGES_REQUESTED** [review]
Hi @prg-04 , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!


## Highlights
✔️ Great commit messages
✔️ PRO PR
✔️ No linter error 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/index.js [hunk 2]
```diff
@@ -1,52 +1,26 @@
...
-    addList.showAlert();
-  } else {
-    const list = new List(input);
-    addList.addList(list);
-    Store.addList(list);
-    addList.clearFields();
-  }
+  const description = document.getElementById('input').value;
+  const list = new List(description);
+  task.addList(list);
```
Comment (ghost) [inline src/index.js]: - [ ] Currently, the problem with deleting items from your todo app persists, kindly check out this [video](https://www.youtube.com/watch?v=AcUd-_Yjjqg) to guide  you on how the functionality works

**Review by frank1738 — CHANGES_REQUESTED** [review]
Hi @prg-04 , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!



## Highlights

✅ Correct use of Github workflow
✅ No linter error


## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

N/A
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by enniopo** [comment]
Hi @prg-04 your dashboard has been updated and you can now request another code review once you are done implementing the changes requested by  @frank1738 . 
Apologies for any inconvenience caused.

File: src/task.js [hunk 3]
```diff
@@ -76,6 +86,14 @@ class Task {
...
   }
 
+  showAlert() {
+    this.alertDiv.textContent = 'Please enter a description for the list.';
+    this.alertDiv.style.display = 'flex';
+    setTimeout(() => {
+      this.alertDiv.style.display = 'none';
+    }, 2000);
+  }
+
```
Comment (koachcode) [inline src/task.js:96]: - Hi @prg-04 , I noticed, the change requested by the previous code reviewer was not implemented, as the window freezes when ever i try performing delete item is operation. 

- I went through your code base and discovered the source of the issue, is in your `for loop` you incremented your `i` as thus `i + 1` which results in an error but changing it to `i += 1` fixes the issue. Please make the appropriate changes to your code base. 

```javascript
 removeList(id) {
    const index = this.items.findIndex((item) => item.id === id);
    if (index !== -1) {
      this.items.splice(index, 1);
      for (let i = index; i < this.items.length; i += 1) {
        this.items[i].id = i + 1;
      }
      this.store.setItems(this.items);
      this.updateList();
    }
  }
```

**Review by koachcode — CHANGES_REQUESTED** [review]
Hi @prg-04 , 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.

keep-trying-youre-almost-there

 ## Highlights
- Linters passed :100: 
- Good file organization :+1: 

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by tufoinnkuo10 — APPROVED** [review]
## Status: Approved :heavy_check_mark:

Hi @prg-04 

Your project is complete! Great job for implementing the changes 🥇. There is nothing else to say other than... it's time to merge it 🍾🚢 :shipit: . Congratulations!  🎉💯🌟 

## Good Points 👍
- Descriptive Pull Request
- CRUD  methods ok
- No linter errors

<br>
Cheers..Happy coding!..and keep soaring higher! 💻 🍷 🚀 

tufoin

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have 2 more limited reviews per this project. If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**